src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -69,28 +84,52 @@ internal async Task InitializeRpcServerAsync()
 
         internal Task InitializeChannelsAsync()
         {
-            string workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
-            if (_environment.IsLinuxAppServiceEnvironment())
+            if (ShouldStartInPlaceholderMode())
             {
-                return Task.CompletedTask;
+                return InitializePlaceholderChannelsAsync();
             }
-            if (_environment.IsLinuxContainerEnvironment())
+            else
             {
-                return Task.CompletedTask;
+                return InitializeWebHostRuntimeChannelsAsync();
             }
-            if (string.IsNullOrEmpty(workerRuntime) && _environment.IsPlaceholderModeEnabled())
+        }
+
+        private Task InitializePlaceholderChannelsAsync()
+        {
+            if (_environment.IsLinuxHostingEnvironment())
             {
-                // Only warm up language workers in placeholder mode in worker runtime is not set
-                return Task.WhenAll(_languages.Select(runtime => _languageWorkerChannelManager.InitializeChannelAsync(runtime)));
+                InitializePlaceholderChannelsAsync(OSPlatform.Linux);
```

Comment:
remove 'return Task.completed` since InitializePlaceholderChannelsAsync returns a Task change to `retrurn InitializePlaceholderChannelsAsync`